common/src/main/java/com/google/auto/common/BasicAnnotationProcessor.java

```diff
@@ -538,9 +538,14 @@ private ElementName(Kind kind, Name name) {
     }
 
     /**
-     * An {@link ElementName} for an annotated element. If {@code element} is a package, uses the
-     * fully qualified name of the package. If it's a type, uses its fully qualified name.
-     * Otherwise, uses the fully-qualified name of the nearest enclosing type.
+     * An {@link ElementName} for an annotated element.
+     *
+     * <p>Packages declared and annotated in {@code package-info.java}, and type elements are tracked
```

Comment:
I'm not sure I agree with this wording. If a package _isn't_ explicitly declared in `package-info.java` we will still track it directly. We won't track it via its nearest enclosing type, which doesn't exist. I think I would prefer to keep the existing text but also add another paragraph, like this:

> A package can be annotated if it has a `package-info.java` with annotations on the `package` declaration.

Reply:
Hmm, if a package isn't declared in `package-info.java` then it can't be annotated, so this method won't be called. Still, I think I prefer my suggestion here.

Reply:
Thank you for the feedback, changed it accordingly.